### PR TITLE
Rework random teams to only perform 2 iterations

### PR DESCRIPTION
**Pull request by scheibo**
After this, the only remaining restrictions on the second pass are:

1) Species clause
2) 6th slot Zoroak
3) bellydrum/swordsdance/calmmind/nastyplot sets
4) multiple zmoves

Not sure if we want to remove the restrictions from 2-4 as well, LMK.

### CODE REVIEW

**Comment by Zarel** [comment]
Why is 6th slot Zoroark a restriction, rather than just a thing you do after generating the team?

**Comment by scheibo** [comment]
> Why is 6th slot Zoroark a restriction, rather than just a thing you do after generating the team?

_Je sais pas_

File: data/random-teams.js [hunk 1]
```diff
@@ -1835,9 +1835,16 @@ class RandomTeams extends Dex.ModdedDex {
 				pokemon.push(set);
 
 				if (pokemon.length === 6) {
-					// Set Zoroark's level to be the same as the last Pokemon
 					let illusion = teamDetails['illusion'];
```
Comment (Zarel) [inline data/random-teams.js]: Why isn't all this just after the `break`?

File: data/random-teams.js [hunk 2]
```diff
@@ -1835,9 +1835,16 @@ class RandomTeams extends Dex.ModdedDex {
...
-					if (illusion) pokemon[illusion - 1].level = pokemon[5].level;
+					if (illusion) {
+						// Swap Zoroark's slot if it is the last Pokemon
+						if (illusion === 5) {
```
Comment (Zarel) [inline data/random-teams.js]: I'm confused, what's going on here? Can't Zoroark be in any slot?

Comment (scheibo) [reply data/random-teams.js]: *if* it happens to be in the last slot, swap it with the second-to-last slot so that Illusion actually *does* something

File: data/random-teams.js (hunk 1 above)
Comment (scheibo) [reply data/random-teams.js]: I don't know, man. I'm not in this part of the codebase to ask questions, I'm here to fix crashes and help out the poor souls stuck with a 5-mon monoice team

File: data/random-teams.js (hunk 2 above)
Comment (Zarel) [reply data/random-teams.js]: Ohhhh. Maybe that comment could be rewritten to be clearer.

File: data/random-teams.js (hunk 1 above)
Comment (Zarel) [reply data/random-teams.js]: I mean, fair, but since you're rewriting this block anyway, you might as well move it to be after the iteration. The less heavily-indented code we have, the better.

File: data/random-teams.js [hunk 3]
```diff
@@ -1835,9 +1835,16 @@ class RandomTeams extends Dex.ModdedDex {
 				pokemon.push(set);
 
 				if (pokemon.length === 6) {
-					// Set Zoroark's level to be the same as the last Pokemon
 					let illusion = teamDetails['illusion'];
-					if (illusion) pokemon[illusion - 1].level = pokemon[5].level;
+					if (illusion) {
+						// Swap Zoroark's slot if it is the last Pokemon
```
Comment (Zarel) [inline data/random-teams.js]:
```suggestion
						// Make sure Zoroark isn't in the last slot
						// (It can't use Illusion if it switches in from the last slot.)
```

File: data/random-teams.js [hunk 4]
```diff
@@ -1869,12 +1866,21 @@ class RandomTeams extends Dex.ModdedDex {
...
+			// Make sure Zoroark isn't in the last slot
+			// (It can't use Illusion if it switches in from the last slot)
+			if (illusion === 5) {
+				[pokemon[5], pokemon[4]] = [pokemon[4], pokemon[5]];
+				illusion = 4;
 			}
+			// Set Zoroark's level to be the same as the last Pokemon
+			pokemon[illusion - 1].level = pokemon[5].level;
 		}
-		if (pokemon.length < 6) throw new Error(`Could not build a random team for ${this.format} (type=${type})`);
```
Comment (Zarel) [inline data/random-teams.js]: Why was this removed? It seems useful to leave in.
